src/bundle/Resources/public/js/scripts/admin.context.menu.js

```diff
@@ -21,40 +25,109 @@
         },
         onAdapted: (visibleItems, hiddenItems) => {
             const hiddenButtonsIds = [...hiddenItems].map((item) => item.querySelector('.ibexa-btn').id);
+            const topBranchItems = multilevelPopupMenu.getBranchItems(topBranch);
 
-            popupMenu.toggleItems((popupMenuItem) => !hiddenButtonsIds.includes(popupMenuItem.dataset.relatedButtonId));
-        },
-    });
-    const popupMenu = new ibexa.core.PopupMenu({
-        popupMenuElement,
-        triggerElement: showPopupButton,
-        onItemClick: (event) => {
-            const { relatedButtonId } = event.currentTarget.dataset;
-            const button = doc.getElementById(relatedButtonId);
+            topBranchItems.forEach((branchItem) => {
+                const shouldBeVisible = hiddenButtonsIds.includes(branchItem.dataset.relatedBtnId);
 
-            button.click();
+                multilevelPopupMenu.toggleItemVisibility(branchItem, shouldBeVisible);
+            });
         },
     });
-    const popupItemsToGenerate = [...menuButtons].map((button) => {
-        const relatedButtonId = button.id;
-        const label = button.querySelector('.ibexa-btn__label').textContent;
+    const clickRelatedBtn = (relatedBtnId) => {
+        const button = doc.getElementById(relatedBtnId);
 
-        return {
-            label,
-            relatedButtonId,
-            disabled: button.disabled,
-        };
+        button.click();
+    };
+    const addRelatedBtnIdToMenuItem = (itemElement, relatedBtnId) => (itemElement.dataset.relatedBtnId = relatedBtnId);
+    const multilevelPopupMenu = new ibexa.core.MultilevelPopupMenu({
+        container: popupMenuElement,
+        triggerElement: showPopupButton,
     });
+    const topBranchItems = menuButtons.map((menuButton) => {
+        const isSplitBtn = menuButton.classList.contains('ibexa-split-btn');
+
+        if (isSplitBtn) {
+            const mainBtn = menuButton.querySelector('.ibexa-split-btn__main-btn');
+            const splitBtn = menuButton.querySelector('.ibexa-split-btn__toggle-btn');
+            const relatedMainBtnId = mainBtn.id;
+            const mainBtnLabel = mainBtn.querySelector('.ibexa-btn__label').textContent;
+            const {
+                alternativeMainBtnLabel: mainBtnAlternativeLabel,
```

Comment:
is it bc you are already using this name in scope?

Reply:
Yes, eslint forced another variable name.

Reply:
I wanted this name to be more consistent with names in the upper scope.